src/python/pants/backend/codegen/tasks/simple_codegen_task.py

```diff
@@ -0,0 +1,155 @@
+# coding=utf-8
+# Copyright 2015 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+
+from pants.backend.core.tasks.task import Task
+from pants.base.address import SyntheticAddress
+from pants.base.build_environment import get_buildroot
+
+
+class SimpleCodegenTask(Task):
+  """Simpler base-class for single-language code-gen.
+
+  Subclasses should implement at minimum: synthetic_target_type, is_gentarget, execute_codegen, and
+  sources_generated_by_target.
+  """
+
+  @classmethod
+  def get_fingerprint_strategy(cls):
```

Comment:
We could add a comment here that says it is optional for subclasses override this method if you want to override the fingerprint strategy in the invalidation check, (but on the other hand, its almost self explanatory.)